lib/postgrex.ex

```diff
@@ -340,9 +341,31 @@ defmodule Postgrex do
     DBConnection.child_spec(Postgrex.Protocol, opts)
   end
 
+  @doc """
+  Returns a stream on a prepared query
+
+  Stream consumes memory in chunks of size `max_rows` at most ( see options )
+
+  This is usefull for processing _large_ datasets
+
+  /!\ this call SHOULD be wrapped in a transaction or face failure on the next chunk /!\
```

Comment:
We can require a `DBConnection.t`, not all `DBConnection.conn`, so we only support inside transaction, i.e. `%DBConnection{} = conn`